File: backend/src/main/java/com/foodgo/entity/ChatMessage.java
package com.foodgo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Table(name = "chat_messages")
@Getter
@Setter
public class ChatMessage {

    public enum Channel { RIDER, CALL_CENTER }
    public enum SenderRole { CUSTOMER, RIDER, CALL_CENTER_AGENT }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "order_id", nullable = false)
    private Long orderId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Channel channel;

    @Column(name = "sender_id", nullable = false)
    private Long senderId;

    @Enumerated(EnumType.STRING)
    @Column(name = "sender_role", nullable = false)
    private SenderRole senderRole;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    @Column(name = "is_read")
    private Boolean isRead = false;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: mobile/lib/models/chat_message.dart
class ChatMessage {
  final int? id;
  final int orderId;
  final String channel; // 'RIDER' or 'CALL_CENTER'
  final int senderId;
  final String senderRole; // 'CUSTOMER' | 'RIDER' | 'CALL_CENTER_AGENT'
  final String message;
  final DateTime? createdAt;

  ChatMessage({
    this.id,
    required this.orderId,
    required this.channel,
    required this.senderId,
    required this.senderRole,
    required this.message,
    this.createdAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json['id'],
      orderId: json['orderId'],
      channel: json['channel'] ?? 'RIDER',
      senderId: json['senderId'],
      senderRole: json['senderRole'] ?? 'CUSTOMER',
      message: json['message'] ?? '',
      createdAt: json['createdAt'] != null ? DateTime.tryParse(json['createdAt']) : null,
    );
  }

  Map<String, dynamic> toSendPayload() => {
        'senderId': senderId,
        'senderRole': senderRole,
        'message': message,
      };
}
